File: quarkus-tmdb/deployment/src/test/java/dev/reuss/tmdb/quarkus/deployment/TmdbHealthCheckRegistrationTest.java
package dev.reuss.tmdb.quarkus.deployment;

import io.quarkus.deployment.Capabilities;
import io.quarkus.deployment.Capability;
import org.junit.jupiter.api.Test;

import java.util.Set;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class TmdbHealthCheckRegistrationTest {

    @Test
    void shouldRegisterHealthCheckWhenSmallRyeHealthIsAvailable() {
        Capabilities capabilities = new Capabilities(Set.of(Capability.SMALLRYE_HEALTH));

        assertTrue(TmdbHealthProcessor.shouldRegisterHealthCheck(capabilities));
    }

    @Test
    void shouldSkipHealthCheckWhenSmallRyeHealthIsMissing() {
        Capabilities capabilities = new Capabilities(Set.of());

        assertFalse(TmdbHealthProcessor.shouldRegisterHealthCheck(capabilities));
    }
}

File: quarkus-tmdb/deployment/src/test/kotlin/dev/reuss/tmdb/quarkus/deployment/TmdbHealthCheckRegistrationTest.kt
package dev.reuss.tmdb.quarkus.deployment

import io.quarkus.deployment.Capabilities
import io.quarkus.deployment.Capability
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class TmdbHealthCheckRegistrationTest {
    @Test
    fun shouldRegisterHealthCheckWhenSmallRyeHealthIsAvailable() {
        val capabilities =
            Capabilities(
                setOf(Capability.SMALLRYE_HEALTH),
            )

        assertTrue(
            TmdbHealthProcessor.shouldRegisterHealthCheck(capabilities),
        )
    }

    @Test
    fun shouldSkipHealthCheckWhenSmallRyeHealthIsMissing() {
        val capabilities =
            Capabilities(
                emptySet(),
            )

        assertFalse(
            TmdbHealthProcessor.shouldRegisterHealthCheck(capabilities),
        )
    }
}
